bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -0,0 +1,1044 @@
+			toolItem.notifyListeners(SWT.Selection, null);
+		}
+
+		@Override
+		public void keyReleased(KeyEvent e) {
+			// Do nothing
+		}
+	};
+
+	private ControlListener shellMovementListener = new ControlListener() {
+		@Override
+		public void controlMoved(ControlEvent e) {
+			positionToPart();
+		}
+
+		@Override
+		public void controlResized(ControlEvent e) {
+			positionToPart();
+		}
+	};
+	private FocusListener overlayFocusListener = new FocusListener() {
+
+		@Override
+		public void focusGained(FocusEvent e) {
+			// do nothing
+		}
+
+		@Override
+		public void focusLost(FocusEvent e) {
+			findReplaceLogic.activate(SearchOptions.GLOBAL);
+			searchInSelectionButton.setSelection(false);
+		}
+
+	};
+	private PaintListener widgetMovementListener = new PaintListener() {
+
+		@Override
+		public void paintControl(PaintEvent e) {
+			positionToPart();
+		}
+
+	};
+	private IPartListener partListener = new IPartListener() {
+		@Override
+		public void partActivated(IWorkbenchPart part) {
+			if (getShell() != null) {
+				getShell().setVisible(isPartCurrentlyDisplayedInPartSash());
+			}
+		}
+
+		@Override
+		public void partDeactivated(IWorkbenchPart part) {
+			// Do nothing
+		}
+
+		@Override
+		public void partBroughtToTop(IWorkbenchPart part) {
+			if (getShell() != null) {
+				getShell().setVisible(isPartCurrentlyDisplayedInPartSash());
+			}
+		}
+
+		@Override
+		public void partClosed(IWorkbenchPart part) {
+			close();
+		}
+
+		@Override
+		public void partOpened(IWorkbenchPart part) {
+			// Do nothing
+		}
+	};
+
+	private boolean isPartCurrentlyDisplayedInPartSash() {
+		IWorkbenchPage activePage = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage();
+
+		// Check if the targetPart is currently displayed on the active page
+		boolean isPartDisplayed = false;
+
+		if (activePage != null) {
+			IWorkbenchPart activePart = activePage.getActivePart();
+			if (activePart != null && activePart == targetPart) {
+				isPartDisplayed = true;
+			}
+		}
+
+		return isPartDisplayed;
+	}
+
+	/**
+	 * Returns the dialog settings object used to share state between several
+	 * find/replace overlays.
+	 *
+	 * @return the dialog settings to be used
+	 */
+	private static IDialogSettings getDialogSettings() {
+		IDialogSettings settings = PlatformUI
+				.getDialogSettingsProvider(FrameworkUtil.getBundle(FindReplaceOverlay.class)).getDialogSettings();
+		return settings;
+	}
+
+	@Override
+	public void create() {
+		if (overlayOpen) {
+			return;
+		}
+		super.create();
+	}
+
+	@Override
+	public boolean close() {
+		if (!overlayOpen) {
+			return true;
+		}
+		findReplaceLogic.activate(SearchOptions.GLOBAL);
+		getDialogSettings().put(REPLACE_BAR_OPEN_DIALOG_SETTING, replaceBarOpen);
+		overlayOpen = false;
+		replaceBarOpen = false;
+		unbindListeners();
+		container.dispose();
+		return super.close();
+	}
+
+	@Override
+	public int open() {
+		int returnCode;
+		if (overlayOpen) {
+			returnCode = Window.CANCEL;
```

Comment:
I don't remember why I returned "CANCEL" - probably because I had to return something and I did not want to look too deeply into what to return.

from the docs (`A window's return codes are
	 * window-specific, although two standard return codes are predefined:
	 * <code>OK</code> and <code>CANCEL</code>.`), I get that this return-code is meant for dialogs which return options to the caller. This Dialog is used as a modal, so it does not make sense to return anything (other than `OK`).